compiler/rustc_metadata/src/rmeta/decoder/cstore_impl.rs

```diff
@@ -304,17 +304,7 @@ pub fn provide(providers: &mut Providers) {
             // traversal, but not globally minimal across all crates.
             let bfs_queue = &mut VecDeque::new();
 
-            // Preferring shortest paths alone does not guarantee a
-            // deterministic result; so sort by crate num to avoid
-            // hashtable iteration non-determinism. This only makes
-            // things as deterministic as crate-nums assignment is,
-            // which is to say, its not deterministic in general. But
-            // we believe that libstd is consistently assigned crate
-            // num 1, so it should be enough to resolve #46112.
-            let mut crates: Vec<CrateNum> = (*tcx.crates(())).to_owned();
-            crates.sort();
-
-            for &cnum in crates.iter() {
+            for &cnum in tcx.crates(()) {
```

Comment:
are these changes just "cleanup" or is there some substantive change in the order in which things are done here?

Reply:
just cleanup, the only change of behavior is the second commit here.

`crates(())` already returns the `cnums` in order, so the code i deleted in the other commits should just be noops